c7n/query.py

```diff
@@ -377,11 +380,13 @@ def get_permissions(self):
             perms.extend(self.permissions)
         return perms
 
-    def resources(self, query=None):
-        key = {'region': self.config.region,
-               'resource': str(self.__class__.__name__),
-               'q': query}
+    def get_cache_key(self, query):
+        return {'region': self.config.region,
```

Comment:
This needs to be extended to include something about the account.  When switching accounts with a number of different mechanisms like profile or assume, you will end up with resources that don't belong to that account.